packages/web3-eth/src/utils/transaction_builder.ts

```diff
@@ -75,7 +76,7 @@ export const getTransactionType = (
 // Keep in mind that the order the properties of populateTransaction get populated matters
 // as some of the properties are dependent on others
 export async function defaultTransactionBuilder<ReturnType = Record<string, unknown>>(options: {
-	transaction: Record<string, unknown>;
```

Comment:
This line should be reverted, the line that was really in question regarding the use of `InternalTransaction` was [this one](https://github.com/ChainSafe/web3.js/blob/6d87ef347c548e875b7cc58df13708f3c1fae2fa/packages/web3-eth/src/utils/transaction_builder.ts#L83)

The reason why this was deemed an issue is because `InternalTransaction` states that the expected transaction properties would be hex strings (for the appropriate properties e.g. `nonce`, `data`, `chainId`, etc.), but of course the use of `Record<string, unknown>` doesn't guarantee that. I don't think we should change the type of the transaction parameter to type `Transaction`, because the type signature of `defaultTransactionBuilder` needs to remain open to any object structure (since developers can provider their own implementation for a transaction builder), so the type should be reverted back to `Record<string, unknown>`

As mentioned, it's this line:

```typescript
let populatedTransaction = { ...options.transaction } as unknown as InternalTransaction;
```

Where we should be performing:

```typescript
format(transactionSchema, populatedTransaction, DEFAULT_RETURN_FORMAT)
```

That way we can enforce that all the expected properties on `transaction` we are to use in our implementation of `defaultTransactionBuilder` are formatted to `DEFAULT_RETURN_FORMAT` (hex strings) before we use them